### PR TITLE
add platform class for querying Java version

### PR DESCRIPTION
**Pull request by richardstartin**

### CODE REVIEW

File: dd-java-agent/src/test/java/jvmbootstraptest/LogManagerSetter.java [hunk 1]
```diff
@@ -211,7 +212,7 @@ private static boolean isTracerInstalled(final boolean wait) {
   }
 
   private static boolean isJavaBefore9WithJFR() {
-    if (!System.getProperty("java.version").startsWith("1.")) {
+    if (!Platform.isJavaVersionAtLeast(9)) {
```
Comment (richardstartin) [inline dd-java-agent/src/test/java/jvmbootstraptest/LogManagerSetter.java:215]: It looks like this method's name was lying

**Review by devinsba — APPROVED** [review]
Works for me. I think `supportsJfr` probably could exist here too with later refactoring

**Comment by richardstartin** [comment]
@devinsba let's leave it up to the profiling team if they want to do that. I'm adding this now because I was about to add another JDK7 check in another branch.

**Comment by lpriima** [comment]
> @devinsba let's leave it up to the profiling team if they want to do that. I'm adding this now because I was about to add another JDK7 check in another branch.

I've added yet another check some time ago too: https://github.com/DataDog/dd-trace-java/blob/master/dd-java-agent/testing/src/main/java/datadog/trace/agent/test/utils/UnsafeUtils.java#L11

We definitely need single place for this check.

File: internal-api/src/test/groovy/datadog/trace/api/PlatformTest.groovy [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+    expected == parsedVersion
+
+    where:
+    version     | expected
+    "1.7"       | 7
+    "1.7.0"     | 7
+    "1.8"       | 8
+    "1.8.0"     | 8
+    "1.8.0_212" | 8
+    "1.8.0_212" | 8
```
Comment (lpriima) [inline internal-api/src/test/groovy/datadog/trace/api/PlatformTest.groovy:38]: duplicate?

looks like java version has a lot : https://openjdk.java.net/jeps/223 :

```
"9-ea" | 9
"9" | 9
"9.1.2" | 9
"9.0.1" | 9
```

**Review by lpriima — APPROVED** [review]

File: internal-api/src/test/groovy/datadog/trace/api/PlatformTest.groovy (hunk 2 above)
Comment (richardstartin) [reply internal-api/src/test/groovy/datadog/trace/api/PlatformTest.groovy]: Nice, thanks.

**Review by bantonsson — APPROVED** [review]

**Pull request merged**